src/Simplify_LT.cpp

```diff
@@ -68,10 +68,18 @@ Expr Simplify::visit(const LT *op, ExprInfo *bounds) {
 
         // clang-format off
         if (rewrite(broadcast(x, c0) < broadcast(y, c0), broadcast(x < y, c0)) ||
+
+            // We can learn more from equality than less with mod.
+            rewrite(x % y < 1, x % y == 0) ||
+            rewrite(0 < x % y, x % y != 0) ||
+            rewrite(x % c0 < c1, x % c0 != fold(c0 - 1), c1 + 1 == c0) ||
```

Comment:
Not convinced this is correct when c0 < 0, or the next rule is correct when c1 < 0